pkg/web/api/connector_v1_test.go

```diff
@@ -203,6 +206,119 @@ func TestConnectorAPIv1_CreateConnector(t *testing.T) {
 	assert.Equal(t, want, got)
 }
 
+func TestConnectorAPIv1_InspectConnector_SendRecord(t *testing.T) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	ctrl := gomock.NewController(t)
+	csMock := apimock.NewConnectorOrchestrator(ctrl)
+	api := NewConnectorAPIv1(csMock)
+
+	id := uuid.NewString()
+	rec := generateTestRecord()
+	recProto, err := toproto.Record(rec)
+	assert.Ok(t, err)
+
+	records := make(chan record.Record)
+
+	csMock.EXPECT().
+		Inspect(ctx, id).
+		Return(records, nil).
+		Times(1)
+
+	inspectServer := apimock.NewConnectorService_InspectConnectorServer(ctrl)
+	inspectServer.EXPECT().Send(gomock.Eq(&apiv1.InspectConnectorResponse{Record: recProto}))
+	inspectServer.EXPECT().Context().Return(ctx).AnyTimes()
+
+	go func() {
+		_ = api.InspectConnector(
+			&apiv1.InspectConnectorRequest{Id: id},
+			inspectServer,
+		)
+	}()
+	records <- rec
+
+	time.Sleep(100 * time.Millisecond)
+}
+
+func TestConnectorAPIv1_InspectConnector_SendErr(t *testing.T) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	ctrl := gomock.NewController(t)
+	csMock := apimock.NewConnectorOrchestrator(ctrl)
+	api := NewConnectorAPIv1(csMock)
+	id := uuid.NewString()
+	records := make(chan record.Record)
+
+	csMock.EXPECT().
+		Inspect(ctx, id).
+		Return(records, nil).
+		Times(1)
+
+	inspectServer := apimock.NewConnectorService_InspectConnectorServer(ctrl)
+	inspectServer.EXPECT().Context().Return(ctx).AnyTimes()
+	errSend := cerrors.New("I'm sorry, but no.")
+	inspectServer.EXPECT().Send(gomock.Any()).Return(errSend)
+
+	errC := make(chan error)
+	go func() {
+		err := api.InspectConnector(
+			&apiv1.InspectConnectorRequest{Id: id},
+			inspectServer,
+		)
+		errC <- err
+	}()
+	records <- generateTestRecord()
```

Comment:
I like this simplification! The thing is, `api.InspectConnector` may block (since it's internally reading from a channel), so by having it in a goroutine we make sure the test doesn't get stuck.